pyccel/codegen/cwrapper.py

```diff
@@ -4,13 +4,18 @@
 from pyccel.ast.core import Module, Declare, Assign
 from pyccel.ast.type_inference import str_dtype
 
+#using the documentation of PyArg_ParseTuple() and Py_BuildValue https://docs.python.org/3/c-api/arg.html
 pytype_registry = {
-        'integer': 'l',
-        'real': 'd',
-        'complex':'c',
-        'bool':'p',
-        'str':'s'
-        }
+    ('real',8)    : 'd',
+    ('real',4)    : 'f',
+    ('complex',8) : 'D',
+    ('complex',4) : 'D',
+    ('int',4)     : 'i',
+    ('int',8)     : 'l',
+    ('int',2)     : 'i',
+    ('int',1)     : 'i',
+    ('bool',4)    : 'i'
```

Comment:
Please revert the boolean and string format types

Reply:
im using (dtype, prec) dict here can you tell me please default string precision

Reply:
or should i go back to the previous py_type dict for the moment.

Reply:
https://github.com/pyccel/pyccel/blob/master/pyccel/ast/core.py#L4747
Apparently string precision is set to 0 as it is a nonsense value for this type